quipucords/api/scanjob/model.py

```diff
@@ -274,3 +287,40 @@ def validate_status_change(self, target_status, valid_current_status):
                          target_status, self.status)
             return True
         return False
+
+    def create_product_status_dict(self):
+        """Construct a dictionary based on the disabled products.
+
+        :returns: a dictionary representing the updated collection
+        status of the optional products
+        """
+        # Grab the optional products status dict and create
+        # a default dict (all products default to True)
+        product_status = self.get_optional_products_args()
+        product_default = {self.OPT_JBOSS_EAP: True,
+                           self.OPT_JBOSS_FUSE: True,
+                           self.OPT_JBOSS_BRMS: True}
+
+        # If specified, turn off fact collection for fuse
+        if not product_status[self.ARG_JBOSS_FUSE]:
```

Comment:
You should generally use the`get` method to avoid a `KeyError` (i.e `product_status.get(self.ARG_JBOSS_FUSE)`, it will return `None` if its not set). Same comment for the lines below.